app/main.py

```diff
@@ -1,3 +1,55 @@
+from __future__ import annotations
+from typing import Union
+
+
 class Distance:
-    # Write your code here
-    pass
+
+    def __init__(self, km: float = 0.0) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    @staticmethod
+    def _convert_to_km(other: Union[int, float, Distance]) -> float:
+        if isinstance(other, Distance):
+            return other.km
+        elif isinstance(other, Union[int, float, Distance]):
+            return float(other)
+        raise TypeError("Unsupported type for arithmetic operation")
+
+    def __add__(self, other: Union[int, float, Distance]) -> "Distance":
+        return Distance(self.km + self._convert_to_km(other))
+
+    def __iadd__(self, other: Union[int, float, Distance]) -> "Distance":
+        self.km += self._convert_to_km(other)
+        return self
+
+    def __mul__(self, other: Union[int, float, Distance]) -> "Distance":
+        return Distance(self.km * self._convert_to_km(float(other)))
+
+    def __truediv__(self, other: (int, float)) -> "Distance":
```

Comment:
`Typing of (int, float) is misleading because it implies that other should be a tuple of int and float, correct way is to use Union[int, float] or use pipe int | float.` -- you didn't fix it here